### PR TITLE
Added solution for Project Euler problem 119

### PR DESCRIPTION
**Pull request by ksr1122**
Name: Digit power sum

Problem Statement: The number 512 is interesting because it is equal to the sum of its digits raised to some power: 5 + 1 + 2 = 8, and 83 = 512. Another example of a number with this property is 614656 = 284.
We shall define an to be the nth term of this sequence and insist that a number must contain at least two digits to have a sum.
You are given that a2 = 512 and a10 = 614656.
Find a30

Reference: https://projecteuler.net/problem=119

reference: #2695

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by ksr1122** [comment]
Review request, @dhruvmanila

File: project_euler/problem_119/sol1.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+"""
+"https://projecteuler.net/problem=119"
```
Comment (dhruvmanila) [inline project_euler/problem_119/sol1.py:2]:
```suggestion
Problem 119: https://projecteuler.net/problem=119
```

File: project_euler/problem_119/sol1.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+Name: Digit power sum
+
+The number 512 is interesting because it is equal to the sum of its digits
+raised to some power: 5 + 1 + 2 = 8, and 83 = 512. Another example of a number
+with this property is 614656 = 284. We shall define an to be the nth term of
+this sequence and insist that a number must contain at least two digits to have a sum.
+You are given that a2 = 512 and a10 = 614656. Find a30
+"""
+
+
```
Comment (dhruvmanila) [inline project_euler/problem_119/sol1.py:13]:
```suggestion
```

File: project_euler/problem_119/sol1.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import math
+
+
+def digit_sum(n: int) -> int:
+    """
+    Returns the sum of the digits of the number
```
Comment (dhruvmanila) [inline project_euler/problem_119/sol1.py:19]: Please include a few `doctest`

File: project_euler/problem_119/sol1.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    """
+    sum = 0
+    while n > 0:
+        sum += n % 10
+        n //= 10
+    return sum
```
Comment (dhruvmanila) [inline project_euler/problem_119/sol1.py:25]:
```suggestion
    return sum([int(digit) for digit in str(n)])
```

File: project_euler/problem_119/sol1.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def solution() -> int:
+    """
+    Returns the value of a30
```
Comment (dhruvmanila) [inline project_euler/problem_119/sol1.py:30]: Please improve on `docstring` and add `doctest`

File: project_euler/problem_119/sol1.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    """
+    powers = []
+    for i in range(2, 100):
+        for j in range(2, 100):
+            n = int(math.pow(i, j))
```
Comment (dhruvmanila) [inline project_euler/problem_119/sol1.py:35]: Can we have a more descriptive variable name than `n`?

File: project_euler/problem_119/sol1.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
+"""
+"https://projecteuler.net/problem=119"
+
+Name: Digit power sum
+
+The number 512 is interesting because it is equal to the sum of its digits
+raised to some power: 5 + 1 + 2 = 8, and 83 = 512. Another example of a number
+with this property is 614656 = 284. We shall define an to be the nth term of
```
Comment (dhruvmanila) [inline project_euler/problem_119/sol1.py:8]:
```suggestion
raised to some power: 5 + 1 + 2 = 8, and 8^3 = 512. Another example of a number
with this property is 614656 = 28^4. We shall define an to be the nth term of
```

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

File: project_euler/problem_119/sol1.py (hunk 1 above)
Comment (ksr1122) [reply project_euler/problem_119/sol1.py]: Changes applied.

File: project_euler/problem_119/sol1.py (hunk 2 above)
Comment (ksr1122) [reply project_euler/problem_119/sol1.py]: removed the empty line.

File: project_euler/problem_119/sol1.py (hunk 3 above)
Comment (ksr1122) [reply project_euler/problem_119/sol1.py]: doctests added

File: project_euler/problem_119/sol1.py (hunk 4 above)
Comment (ksr1122) [reply project_euler/problem_119/sol1.py]: replaced with the suggested simplified version.

File: project_euler/problem_119/sol1.py (hunk 5 above)
Comment (ksr1122) [reply project_euler/problem_119/sol1.py]: updated docstring and added doctests

File: project_euler/problem_119/sol1.py (hunk 7 above)
Comment (ksr1122) [reply project_euler/problem_119/sol1.py]: fixed the docstring.

File: project_euler/problem_119/sol1.py (hunk 2 above)
Comment (dhruvmanila) [reply project_euler/problem_119/sol1.py]: Remove both the newlines before the imports :)
You could've directly applied the suggestion but it's up to you.

File: project_euler/problem_119/sol1.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    512
+    >>> solution(5)
+    5832
+    >>> solution(10)
+    614656
+    """
+    digi_to_powers = []
+    for digit in range(2, 100):
+        for power in range(2, 100):
+            digit_to_power = int(math.pow(digit, power))
```
Comment (dhruvmanila) [inline project_euler/problem_119/sol1.py:42]: Maybe `number` instead of `digit_to_power` and `digit_to_powers` instead of `digi_to_powers` ?

Comment (ksr1122) [reply project_euler/problem_119/sol1.py]: oops, sorry for the typo. Fixed it.

**Review by dhruvmanila — APPROVED** [review]

**Pull request merged**